### PR TITLE
GEODE-7536: Add serialization analysis to geode-gfsh

### PR DESCRIPTION
**Pull request by bschuchardt**
The new geode-gfsh module left all of its serializables in the
sanctioned list from geode-core.  This commit adds serialization
analysis to the new module and removes geode-gfsh serializables from
sanctioned-geode-core-serializables.txt.

I had to modify the base test class to read the sanctioned list using a
resource stream instead of a file.  Geode-gfsh, for some reason, has the
module's JAR file on the path and the test is finding the resource there
instead of as a regular file.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [n/a] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-junit/src/main/java/org/apache/geode/codeAnalysis/AnalyzeSerializablesJUnitTestBase.java [hunk 1]
```diff
@@ -88,7 +89,7 @@
   private File expectedDataSerializablesFile;
   private String expectedSerializablesFileName =
       "sanctioned-" + getModuleName() + "-serializables.txt";
-  private File expectedSerializablesFile;
+  private InputStream expectedSerializablesInputStream;
```
Comment (upthewaterspout) [inline geode-junit/src/main/java/org/apache/geode/codeAnalysis/AnalyzeSerializablesJUnitTestBase.java]: Why is this InputStream a field on the class? Doesn't seem like something we should keep around or reuse.

File: geode-junit/src/main/java/org/apache/geode/codeAnalysis/CompiledClassUtils.java [hunk 2]
```diff
@@ -292,11 +294,10 @@ public static void storeClassesAndMethods(List<ClassAndMethods> cams, File file)
...
   }
 
-  public static List<ClassAndVariableDetails> loadClassesAndVariables(File file)
+  public static List<ClassAndVariableDetails> loadClassesAndVariables(InputStream stream)
       throws IOException {
-    List<ClassAndVariableDetails> result = new LinkedList<ClassAndVariableDetails>();
-    FileReader fr = new FileReader(file);
-    BufferedReader in = new BufferedReader(fr);
+    List<ClassAndVariableDetails> result = new LinkedList<>();
+    BufferedReader in = new BufferedReader(new InputStreamReader(stream));
```
Comment (upthewaterspout) [inline geode-junit/src/main/java/org/apache/geode/codeAnalysis/CompiledClassUtils.java]: Should this be in a try-with-resources? Doesn't look like we put the close in a finally block.

File: geode-junit/src/main/java/org/apache/geode/codeAnalysis/AnalyzeSerializablesJUnitTestBase.java [hunk 3]
```diff
@@ -433,7 +433,7 @@ private void loadClasses() throws IOException {
     String ideaBuildDirName = Paths.get(getModuleName(), "out", "production", "classes").toString();
     System.out.println("ideaBuildDirName is " + ideaBuildDirName);
     String ideaFQCNBuildDirName = Paths.get("out", "production",
-        "org.apache.geode." + getModuleName() + ".main").toString();
+        "geode." + getModuleName() + ".main").toString();
```
Comment (upthewaterspout) [inline geode-junit/src/main/java/org/apache/geode/codeAnalysis/AnalyzeSerializablesJUnitTestBase.java]: I wish there was a way to do this without fishing for an intellij (or gradle) directory name.

Comment (bschuchardt) [reply geode-junit/src/main/java/org/apache/geode/codeAnalysis/AnalyzeSerializablesJUnitTestBase.java]: I've looked at how ArchUnit finds classes and it's also classpath & search based but is considerably more complex, hooking into analysis classes in gradle, idea and in one case a class available starting in jdk 9.

File: geode-junit/src/main/java/org/apache/geode/codeAnalysis/CompiledClassUtils.java (hunk 2 above)
Comment (bschuchardt) [reply geode-junit/src/main/java/org/apache/geode/codeAnalysis/CompiledClassUtils.java]: I changed this to a try-with-resources and moved the closing back to AnalyzeSerializablesJUnitTestBase.

File: geode-junit/src/main/java/org/apache/geode/codeAnalysis/AnalyzeSerializablesJUnitTestBase.java (hunk 1 above)
Comment (bschuchardt) [reply geode-junit/src/main/java/org/apache/geode/codeAnalysis/AnalyzeSerializablesJUnitTestBase.java]: I've deleted this instance variable

**Pull request merged**